src/main/java/utilities/FailureHandler.java

```diff
@@ -0,0 +1,166 @@
+    private final static String recordingVideoName = "Recorded";
+    private static SpecializedScreenRecorder screenRecorder;
+    private static String videosDir = "videos/";
+    private static String pageSourcesDir = "pageSources/";
+    private static String screenshotsDir = "screenshots/";
+
+    public static void takePageSource(Scenario scenario) {
+        String pageSource = Context.driverManager.getDriver().getPageSource();
+
+        String timeToPrint = getCurrentTime();
+
+        Path path = Paths.get(pageSourcesDir + scenario.getName() + "_" + timeToPrint + ".html");
+
+        try {
+            Files.createFile(path);
+            Files.write(path, pageSource.getBytes());
+            scenario.embed(pageSource.getBytes(), "text/html");
+        } catch (IOException e) {
+            log.error("Failed to take screenshot");
+            e.printStackTrace();
+        }
+
+    }
+
+    public static void takeScreenshot(Scenario scenario) {
+        String timeToPrint = getCurrentTime();
+
+        try {
+            File screenSource = ((TakesScreenshot) Context.driverManager.getDriver()).getScreenshotAs(OutputType.FILE);
+            FileHandler.copy(
+                    screenSource,
+                    new File(screenshotsDir + scenario.getName() + "_" + timeToPrint + ".png")
+            );
+            byte[] screenSource2 = ((TakesScreenshot) Context.driverManager.getDriver()).getScreenshotAs(OutputType.BYTES);
+            scenario.embed(screenSource2, "image/png");
+        } catch (Exception e) {
+            log.error("Failed to take screenshot");
+            e.printStackTrace();
+        }
+    }
+
+    public static SpecializedScreenRecorder setUpScreenRecorder() {
+        GraphicsConfiguration gc = GraphicsEnvironment
+                .getLocalGraphicsEnvironment()
+                .getDefaultScreenDevice()
+                .getDefaultConfiguration();
+
+        Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
+        int width = screenSize.width;
+        int height = screenSize.height;
+
+        Rectangle captureSize = new Rectangle(0, 0, width, height);
+
+        try {
+            screenRecorder = new SpecializedScreenRecorder(gc, captureSize,
+                    new Format(MediaTypeKey, MediaType.FILE, MimeTypeKey, MIME_AVI),
+                    new Format(MediaTypeKey, MediaType.VIDEO, EncodingKey, ENCODING_AVI_TECHSMITH_SCREEN_CAPTURE,
+                            CompressorNameKey, ENCODING_AVI_TECHSMITH_SCREEN_CAPTURE,
+                            DepthKey, 24, FrameRateKey, Rational.valueOf(15),
+                            QualityKey, 1.0f,
+                            KeyFrameIntervalKey, 15 * 60),
+                    new Format(MediaTypeKey, MediaType.VIDEO, EncodingKey, "black",
+                            FrameRateKey, Rational.valueOf(30)),
+                    null, new File("videos/"), recordingVideoName);
```

Comment:
didn't check this one yet but I think we should avoid slashes & backslashes at all cost. Because of different filesystems behaviors